### PR TITLE
configurable tmp dir

### PR DESCRIPTION
**Pull request by chrstnbwnkl**
### Proposed Changes

This PR separates the directories for created packages and all other files (i.e. planet PBF, graph, etc.). `DATA_DIR` refers to the location where all created packages will be stored whereas `TMP_DIR` stores the planet PBF, the graph tiles, elevation data, and so on.

### Migration

Migration should be easy, just move all the files that are not packaged extracts to the new `TMP_DATA_DIR` location before starting the services.

### CODE REVIEW

**Comment by chrstnbwnkl** [comment]
we can run the test once the update only PR is merged and we change the target branch to main :)

File: docker-compose.yml [hunk 1]
```diff
@@ -4,8 +4,13 @@ volumes:
     driver: local
     driver_opts:
       type: none
-      device: $DATA_DIR  # DATA_DIR is the host directory for the data. It has to be in the environment, e.g. in .env file
+      device: $DATA_DIR # DATA_DIR is the host directory for the data. It has to be in the environment, e.g. in .env file
       o: bind
+  tmp_packages: # do not change any detail of this volume
```
Comment (chrstnbwnkl) [inline docker-compose.yml:9]: I just copied the volume configuration from the other one for now.

File: docker-compose.yml [hunk 2]
```diff
@@ -4,8 +4,13 @@ volumes:
...
+    driver: local
+    driver_opts:
+      type: none
+      device: $TMP_DATA_DIR # DATA_DIR is the host directory for the data. It has to be in the environment, e.g. in .env file
```
Comment (nilsnolde) [inline docker-compose.yml:13]:
```suggestion
      device: $TMP_DATA_DIR # TMP_DATA_DIR is the host directory for the data. It has to be in the environment, e.g. in .env file
```

File: docker-compose.yml (hunk 1 above)
Comment (nilsnolde) [reply docker-compose.yml:9]: let's call it just `tmp_data`, there's no packages

**Comment by nilsnolde** [comment]
hm, I just realize CI is still running on ubuntu 20.04. might wanna change that to ubuntu-latest (either here or at some point).

**Comment by nilsnolde** [comment]
It might also be good to have a test in place. Current tests are checking whether a package was created and has the right graph packaged, but all from a request's response. I guess it'd be enough to check the path set by the env var?

**Comment by nilsnolde** [comment]
> hm, I just realize CI is still running on ubuntu 20.04. might wanna change that to ubuntu-latest (either here or at some point).

was thinking about the other PR :sweat_smile:

**Comment by chrstnbwnkl** [comment]
> It might also be good to have a test in place. Current tests are checking whether a package was created and has the right graph packaged, but all from a request's response. I guess it'd be enough to check the path set by the env var?

End to end testing is not trivial, we don't actually push the jobs in the Arq queue when testing. This would require us to have a distinct test setup with only a small PBF instead of the planet and so on... maybe something for the future.

**Pull request merged**